pkg/input/input.go

```diff
@@ -0,0 +1,26 @@
+package input
+
+import (
+	"bufio"
+	"os"
+)
+
+type Directory interface {
```

Comment:
all of those things already seem to exist in fs.go in the fs package, why do we need to reimplement that?

Reply:
There's a `DirEntry` interface but it's not the same. I can rename the interface if you think it's confusing. I've refactored the structs to use some of interfaces but it's not a 1:1 match.

Reply:
it's alright, the more we can reuse the better - in case we need other standard lib functionality that uses those interfaces. Thanks!